python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -39,14 +39,13 @@
     from ..schema.util import ResolveInfo
 
 
-def get_run_by_id(
+async def gen_run_by_id(
     graphene_info: "ResolveInfo", run_id: str
 ) -> Union["GrapheneRun", "GrapheneRunNotFoundError"]:
     from ..schema.errors import GrapheneRunNotFoundError
     from ..schema.pipelines.pipeline import GrapheneRun
 
-    instance = graphene_info.context.instance
-    record = instance.get_run_record_by_id(run_id)
+    record = await graphene_info.context.get_loader_for(RunRecord).load(run_id)
```

Comment:
might be better to instead do `RunRecordLoader.some_method(context, run_id)` and have that static method manage interacting with the request contexts stored loaders

Reply:
Agree.

IMO the dominant class should be the class that gets return, rather than the loader, which should be generally abstracted away from the User.

e.g.

`RunRecord.load(context, run_id)`

And some magic should get the loader for you.

Also a good question of what the context should be. I think it will be hard to answer right now but we should future proof it as much as possible to make it easy to change what the context object is. I'm thinking in particular of "hoisting" the `TemporalContext` object out of the `AssetGraphView` layer up to here.

Reply:
> RunRecord.load(context, run_id)

well, this seems obvious in hindsight. Not sure how i didn't get all the way there

Reply:
I think the other thing to consider is adding a base class to `RunRecord` so can handle `gen_` method on those instances. 

It should also probably be `RunRecord.gen` and maybe a `RecordRecord.gen_many` for pluralized.